### DISCUSSION TITLE
Plex Shares

### DISCUSSION THREAD

**Discussion by umbrellaplug** [umbrellaplug/umbrellaplug.github.io#60] [General]
Plex shares are collections of content that people have shared using plex.tv. 

Plex shares can be found on Reddit in the r/plexshares subreddit. 

There are several types of plex shares. Free to access, pay to access, and share for share. 

You do not need any type of paid account with plex for this. You are able to access plex and plex shares with just a free plex.tv account. They also sell a plex pass to allow for extras like streaming to your phone or from the web. 

In Umbrella you can add plex shares as sources to be used when scraping for content. Some plex shares have thousands of items of content and could easily serve as a backup service to debrid or Usenet.

**Plex is authorized in CocoScrapers**

Once you authorize plex.tv in the scrapers the plugin will then cycle through your available shares till you select one. 

I am not associated with plex in any way and cannot provide any support for their product. If you have issues with plex outside of Umbrella please contact plex. 

Some servers may kick you off after 10 minutes, if you find this is an issue for you there is an addon called "composition" that will prevent this from happening when used with Umbrella.

**Comment by TheGuyWhoo** [comment]
This is awesome!! I  would very much like to use multiple servers as sources. Is this something that is already possible or that you might implement? Support for transcoding would also be very handy to have.

**Comment by TheGuyWhoo** [comment]
Omg I see you can already transcode if you have Composite installed... Insane! I've been wanting an addon where I can have all my shit in one place... if I could use multiple servers as sources this would be the perfect setup.

**Reply by umbrellaplug** [reply]
This is already setup to scrape all servers you have access to on the username you authorize with unfortunately right now we're only scraping one at a time. Whichever you have set. I will see about your request though.

**Reply by TheGuyWhoo** [reply]
Yes I mean I would like to be able to set more than one. Thank you, I appreciate that!

**Comment by ad1002549** [comment]
My plex account has 2 servers connected. I only can select the first it didn't give the option to choose which server to pick

**Reply by umbrellaplug** [reply]
When you get the popup for setting shares. Select no on the first one and the additional one will popup. I'm working on a scraper update to allow multiple shares to be scraped also.

**Comment by diimaan** [comment]
How do I use composite with Umbrella? I have the Composite addon for plex installed but I am not sure to use that in conjunction with Umbrella! Appreciate any help!

**Reply by umbrellaplug** [reply]
Composite is used automatically if detected when opening Plex links.

**Reply by diimaan** [reply]
Got that! My problem was it is not detecting any server! When I do select(reselect) plexshare source, it asks me about first two server shares on my list and when I select no it has no server selected in "plexshare source title"! Unfortunately I have more than 4 shared sources on my account which are accessible through composite but not through cocoscraper!

**Reply by umbrellaplug** [reply]
Hmm. Don't think I've ever tried more than two servers. I'll see if the scraper dev can take a look at that.

**Reply by diimaan** [reply]
Thank you! Yes I am an anomaly in PlexShare world, I like to experiment with different servers with friends and as well as *so called blockable* P4S shares. :) I really appreciate your effort and time on the addon for Kodi

**Comment by TheGuyWhoo** [comment]
I see you have completely removed issues tab. I wonder why?

**Reply by umbrellaplug** [reply]
Issues tab is just closed for a bit. I've had a few discouraging interactions recently and taking a bit of a break to avoid making it "not fun anymore".

I'll open them back up after i clear some.

**Reply by TheGuyWhoo** [reply]
Oh wow, I'm sorry to hear that. Any reasonable person will understand that you do this for free and that it's a luxury to have a dev as receptive and accommodating as you, I bet it isn't always easy either. If you were to decide to focus your efforts on supporting skins, etc. that are less of a headache that would totally be fair. But in any case I think most of the kodi community who is into this sort of thing views you and your plugin very favorably, it's definitely my favorite.

**Comment by umbrellaplug** [comment]
@diimaan @ad1002549 @TheGuyWhoo 
Any of you have a couple plex shares I can get added to so I can work on this multi share scrape?

**Reply by TheGuyWhoo** [reply]
Yes I can help you out with some discord invites. I don't personally run any server, though. Also reddit allows you to search by flair, so you can easily find free servers that way on r/plexshares(You just type flair:free and limit your search results to the sub).

**Reply by umbrellaplug** [reply]
I was able to get added to a few free ones yesterday so I could test for the scraper dev. It's all good now.

**Comment by umbrellaplug** [comment]
![Screen Shot 2023-01-11 at 5 42 35 PM (2)](https://user-images.githubusercontent.com/107957815/211942406-183c09d8-324d-4645-9700-a937358a222e.png)

Coming in hot.....

**Comment by diimaan** [comment]
Sorry for dusting this up again. Even with the latest scraper and umbrella versionsI am not able to select multiple sources! The only share that I can select is my local share running through pms. There is no other share showing up if I select select/re select source.
This is a on a fresh nexus installation on mac os ventura. I believe I have cleared most of the kodi data and started fresh. I might have missed some of the remnants of addon data. Let me know should I try something else. May be I am doing something wrong here

![coco-plex](https://user-images.githubusercontent.com/3683925/213989560-d2163d0f-4528-4f99-991d-d8b6460d22ce.jpg)

**Reply by umbrellaplug** [reply]
Do you have a discord or a way I can message you outside of here? I'd like to send you a special coco package that logs your headers when making the plex request. I believe it's likely something to do with that. I'm sure you don't want to post your headers here and I do want to post coco here. If you can give me a way to contact you, I'll send you a link to download a special version.

@diimaan

**Reply by diimaan** [reply]
@umbrellaplug 
Yes, I do have a discord. Is there a channel for umbrella? Or can I dm you? diimaan#6879

**Reply by umbrellaplug** [reply]
umbrella_dev#7132

**Comment by dabutch3r** [comment]
Just noticed this feature. I authorize fine through plex.tv/link but one of the servers I use daily is not showing.  I tried reauthorization several times, readd servers button and even looked on my pc plex server and the hidden server in kodi is showing on other devices. I even pinned the server thinking that might help

**Comment by umbrellaplug** [comment]
Some servers may not show up. If the server is not a plex.direct supported server it will not show.

**Reply by dabutch3r** [reply]
Ya it's direct play but not showing.  Odd how all other servers of mine show up. I'm talking with the guy that has that server to figure out the hang up

**Reply by umbrellaplug** [reply]
Let me know if you need information for what's being looked at for this to provide the person maintaining the server.

**Reply by dabutch3r** [reply]
If you could get up with me on reddit I can  try to help

**Reply by umbrellaplug** [reply]
I don't have chat or messages turned on for Reddit. It gets overwhelming.

**Reply by dabutch3r** [reply]
Keybase,  discord or telegram? If you have or get those you can track me down.  I had a guy with the plex server that wasn't showing up, wanting to try to help

**Reply by umbrellaplug** [reply]
discord - umbrella_dev#7132

**Reply by umbrellaplug** [reply]
Plex was updated in the last several updates to allow for more servers to be added. You should be able to add 99% of servers now.

**Comment by dabutch3r** [comment]
The one server uses dns and cloud flare not a direct ip

Todd Buckner

**Reply by umbrellaplug** [reply]
Coco Hits this URL:
[https://plex.tv/api/v2/resources?includeHttps=1&X-Plex-Client-Identifier="CLIENT"&X-Plex-Token="TOKEN](https://plex.tv/api/v2/resources?includeHttps=1&X-Plex-Client-Identifier=%22CLIENT%22&X-Plex-Token=%22TOKEN)"

That client-id and token come from when you authorize plex.

In the response it gets it's looking for: connections. From those connections it pulls "uri" and "local" if ".plex.direct:" is not in the uri OR local is set to 1, it does not give that server as an option. I can give you a special version of coco with logging turned on to see what you are getting back there if you need. Also, I didn't write this so if something needs to be changed for the plex server detection I'm open to suggestions.

**Reply by umbrellaplug** [reply]
Plex was updated in the last several updates to allow for more servers to be added. You should be able to add 99% of servers now.

**Comment by billyneptune** [comment]
Does transcoding work with this?  When umbrella is playing a movie with the source being my intel nuc server, I'm checking plex dashboard and see network activity, but no actual "now playing" event.

**Reply by umbrellaplug** [reply]
Transcoding would be done in the server as far as I understand it. I believe that's the reason he was looking for Plex direct in the url when he originally wrote the Plex scraper because we need a source that's playing direct and not transcoding. As for the activity part, I'm not sure. I will check. I've never checked if my plex server showed me watching when playing from Kodi.

**Reply by umbrellaplug** [reply]
Wanted to reply I looked into this more and tried it. My server does not show any activity in dashboard when playing direct like this. I'd love to find someone that knew more about how plex worked to get some answers on the differences here.

**Reply by billyneptune** [reply]
Yes I have the same outcome.

Another weird thing: when I have a Vpn disconnected on my shield (kodi client) and play a movie through my plex share server, the network activity in my plex dashboard shows "local" but doesn't give full speed and buffers the movie alot.  If I connect to a Vpn on my shield kodi client and do the same thing, I now get "remote" network activity and get full speeds with no buffering.

Example: 
Play 4k 80gb movie through plex share on kodi client with vpn disconnected.
Result: max 30mbs of "local" network a activity show in plex dashboard. Movie buffers alot.

Play same 4k 80gb movie with vpn connected
Result: Roughly 70mbs of "remote" network activity on plex dashboard.  No movie buffering.

note- I have gigabit internet, my plex server runs with no hiccups through a plex client with same 4k movie, both local and remotely

**Reply by umbrellaplug** [reply]
> Yes I have the same outcome.
> 
> 
> 
> Another weird thing: when I have a Vpn disconnected on my shield (kodi client) and play a movie through my plex share server, the network activity in my plex dashboard shows "local" but doesn't give full speed and buffers the movie alot.  If I connect to a Vpn on my shield kodi client and do the same thing, I now get "remote" network activity and get full speeds with no buffering.
> 
> 
> 
> Example: 
> 
> Play 4k 80gb movie through plex share on kodi client with vpn disconnected.
> 
> Result: max 30mbs of "local" network a activity show in plex dashboard. Movie buffers alot.
> 
> 
> 
> Play same 4k 80gb movie with vpn connected
> 
> Result: Roughly 70mbs of "remote" network activity on plex dashboard.  No movie buffering.
> 
> 
> 
> note- I have gigabit internet, my plex server runs with no hiccups through a plex client with same 4k movie, both local and remotely

I'm honestly not sure how I could help with that. Sounds like a network config situation. The scraper never uses your local address for Plex. It's always going to use the external ip.

**Reply by billyneptune** [reply]
Would you be able to test this on your end with vpn on and then off?  I'm curious if it changes between local and remote like it does for me.

**Reply by umbrellaplug** [reply]
I'm not sure I would be able to. I don't use a VPN. I could give you a version of coco with additional logging that gives the addresses being used for playing so you could try to compare.

**Reply by billyneptune** [reply]
Well when you plex share a movie onto your kodi shield client, does it show remote or local in the plex dashboard network activity?

**Reply by billyneptune** [reply]
Update:  So I noticed when I turn my vpn off and try to open kodi (ah2+umbrella+tmdb helper), it gets stuck on "waiting for plugins" for about 1 minute before it finally loads in.  When I have my VPN turned on it loads in like 5 seconds.  The heck is going on?

**Reply by billyneptune** [reply]
Another update:

So I finally got passed the "waiting for plugins" problem above, by connecting with a VPN and revoking the trakt api in both tmdb helper and umbrella.

Now I'm able to load in without vpn, though now there is another problem:

When clicking on a movie poster to scrape for providers, it sits at loading for a really long time.  It'll take around a minute for the source list to appear so I can pick my source.

This problem occurs with my home internet without vpn connected only.  If I connect to a VPN, it scrapes instantly and I can see the sources in about 3 seconds.  If I connect my shield to my cell phone mobile hotspot, it also scrapes instantly.

What the heck could be the issue?

My home internet speeds are around 300mbps with and without vpn.

**Reply by umbrellaplug** [reply]
Make sure ipv6 is turned off on your shield.

**Reply by billyneptune** [reply]
Ok that fixed it...

Wish I knew that before factory resetting my shield but oh well.  Thanks sir I appreciate it greatly.

**Reply by billyneptune** [reply]
Last issue hopefully:

Can you check when you plex share a movie onto your kodi client, does it show remote or local in the plex dashboard network activity?  This issue happened both on my windows pc and nvidia shield.

For  me when I'm on the same lan as my plex client, it shows local activity, and doesn't get up to normal speeds which causes bad buffering.  Only thing that fixes that is turning on a VPN.

**Reply by umbrellaplug** [reply]
These are connection issues. None of this has to do with umbrella or the scrapers. This entriely related to r
Plex and your connection. Please check with Plex about your server configuration for issues with different connections.

**Reply by billyneptune** [reply]
Are you able to test it on your end?   Plex share a movie with your kodi client, and check the plex dashboard to see if it's local or remote.

![Screenshot 2023-03-20 072703](https://user-images.githubusercontent.com/27868673/226354012-0adbeb55-ab3c-420f-8776-80022c9f69f2.png)

**Reply by umbrellaplug** [reply]
Plex was updated in the last several updates to allow for more servers to be added. You should be able to add 99% of servers now.

**Comment by tony2501** [comment]
It works super well but sometimes it does not return a search successfully, saying no content found, when I know i have that content on my plex. it seems to be hit and miss? Is there a known fix?
